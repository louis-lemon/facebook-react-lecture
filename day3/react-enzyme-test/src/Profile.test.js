import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile />', () => {
    it('mathces snapshot', () => {
        const wrapper = mount(<Profile username="louis" name="tak" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders username and name', () => {
        const wrapper = mount(<Profile username="louis" name="tak" />);
        expect(wrapper.props().username).toBe('louis');
        expect(wrapper.props().name).toBe('tak');

        const boldElement = wrapper.find('b');
        expect(boldElement.contains('louis')).toBe(true);

        const spanElement = wrapper.find('span');
        expect(spanElement.text()).toBe('(tak)');
    });
});