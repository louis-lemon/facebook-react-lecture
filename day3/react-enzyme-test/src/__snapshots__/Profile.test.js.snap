// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> mathces snapshot 1`] = `
<Profile
  name="tak"
  username="louis"
>
  <div>
    <b>
      louis
    </b>
    Â 
    <span>
      (
      tak
      )
    </span>
  </div>
</Profile>
`;
